src/Sylius/Bundle/PromotionBundle/Form/DataTransformer/MoneyIntToLocalizedStringTransformer.php

```diff
@@ -17,19 +17,19 @@
 
 final class MoneyIntToLocalizedStringTransformer extends MoneyToLocalizedStringTransformer
 {
-    public function reverseTransform($value)
+    public function reverseTransform(mixed $value): int|float|null
     {
         if (!is_numeric($value)) {
-            return;
+            return null;
         }
 
         return (int) parent::reverseTransform($value);
     }
 
-    public function transform($value)
+    public function transform(mixed $value): string
     {
         if (!is_numeric($value)) {
-            return;
+            return '';
```

Comment:
Could be a BC break, a property could be nullable in the model data, what about returning `null` and type hinting `: ?string` instead?

Reply:
that would violate Symfony 6 contract :/ but! what about fall backing to the Symfony logic instead? In Sf 5.4 we would return the current value and they will handle the return value switch for us? nah... that will break the same case

Reply:
Thinking again about it, I may be wrong.
There would be no BC break as a nullable property would be type hinted `?string` thus accepting a strict string return from here (hopefully a `NotBlank` constraint may help to behave the same).

Reply:
as this class is final, adding returning typehint was not a BC Break - it was an acceptable solution anyway. Returning an empty string instead of a null is changing the class logic, which does not fall directly under BC Policy. Thanks for rising your tensions, tho! It was good to give it a second thoughts